test/e2e/apimachinery/validatingadmissionpolicy.go

```diff
@@ -371,6 +376,415 @@ var _ = SIGDescribe("ValidatingAdmissionPolicy [Privileged:ClusterAdmin]", func(
+	   The validatingadmisionpolicy resource must support create, get,
+	     list, watch, update, patch, delete, and deletecollection.
+	*/
+	framework.ConformanceIt("should support ValidatingAdmissionPolicy API operations", func(ctx context.Context) {
+		vapVersion := "v1"
+		ginkgo.By("getting /apis")
+		{
+			discoveryGroups, err := f.ClientSet.Discovery().ServerGroups()
+			framework.ExpectNoError(err)
+			found := false
+			for _, group := range discoveryGroups.Groups {
+				if group.Name == admissionregistrationv1.GroupName {
+					for _, version := range group.Versions {
+						if version.Version == vapVersion {
+							found = true
+							break
+						}
+					}
+				}
+			}
+			if !found {
+				framework.Failf("expected ValidatingAdmissionPolicy API group/version, got %#v", discoveryGroups.Groups)
+			}
+		}
+
+		ginkgo.By("getting /apis/admissionregistration.k8s.io")
+		{
+			group := &metav1.APIGroup{}
+			err := f.ClientSet.Discovery().RESTClient().Get().AbsPath("/apis/admissionregistration.k8s.io").Do(ctx).Into(group)
+			framework.ExpectNoError(err)
+			found := false
+			for _, version := range group.Versions {
+				if version.Version == vapVersion {
+					found = true
+					break
+				}
+			}
+			if !found {
+				framework.Failf("expected ValidatingAdmissionPolicy API version, got %#v", group.Versions)
+			}
+		}
+
+		ginkgo.By("getting /apis/admissionregistration.k8s.io/" + vapVersion)
+		{
+			resources, err := f.ClientSet.Discovery().ServerResourcesForGroupVersion(admissionregistrationv1.SchemeGroupVersion.String())
+			framework.ExpectNoError(err)
+			foundVAP, foundVAPStatus := false, false
+			for _, resource := range resources.APIResources {
+				switch resource.Name {
+				case "validatingadmissionpolicies":
+					foundVAP = true
+				case "validatingadmissionpolicies/status":
+					foundVAPStatus = true
+				}
+			}
+			if !foundVAP {
+				framework.Failf("expected validatingadmissionpolicies, got %#v", resources.APIResources)
+			}
+			if !foundVAPStatus {
+				framework.Failf("expected validatingadmissionpolicies/status, got %#v", resources.APIResources)
+			}
+		}
+
+		client := f.ClientSet.AdmissionregistrationV1().ValidatingAdmissionPolicies()
+		labelKey, labelValue := "example-e2e-vap-label", utilrand.String(8)
+		label := fmt.Sprintf("%s=%s", labelKey, labelValue)
+
+		template := &admissionregistrationv1.ValidatingAdmissionPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-example-vap-",
+				Labels: map[string]string{
+					labelKey: labelValue,
+				},
+			},
+			Spec: admissionregistrationv1.ValidatingAdmissionPolicySpec{
+				Validations: []admissionregistrationv1.Validation{
+					{
+						Expression: "object.spec.replicas <= 100",
+					},
+				},
+				MatchConstraints: &admissionregistrationv1.MatchResources{
+					ResourceRules: []admissionregistrationv1.NamedRuleWithOperations{
+						{
+							RuleWithOperations: admissionregistrationv1.RuleWithOperations{
```

Comment:
do we need to add something to this rule so it doesn't disrupt unrelated deployments (some namespace selector of object selector), or is this inert because we're not creating a binding?